src/main/scala/strawman/collection/Iterable.scala

```diff
@@ -475,7 +475,18 @@ trait IterableOps[+A, +CC[X], +C] extends Any {
   def take(n: Int): C = fromSpecificIterable(View.Take(coll, n))
 
   /** A collection containing the last `n` elements of this collection. */
-  def takeRight(n: Int): C = fromSpecificIterable(View.TakeRight(coll, n))
+  def takeRight(n: Int): C = {
+    val b = newSpecificBuilder()
+    b.sizeHintBounded(n, coll)
+    val lead = coll.iterator() drop n
+    val it = coll.iterator()
+    while (lead.hasNext) {
+      lead.next()
```

Comment:
I tried it but didn’t notice a difference (in `List`):

![takeright](https://user-images.githubusercontent.com/332812/28369423-d1f9d9c2-6c97-11e7-8542-3deb1ce446ce.png)